File: python/bob/bob.py
def response(hey_bob: str) -> str:
    hey_bob = hey_bob.strip()

    if not hey_bob: return "Fine. Be that way!"

    if hey_bob[-1] == "?":
        # If hey_bob is a question.
        if hey_bob.isupper():
            return "Calm down, I know what I'm doing!"

        return "Sure."

    if hey_bob.isupper():
        return "Whoa, chill out!"

    return "Whatever."

File: csharp/bob/Bob.cs
using System;

public static class Bob
{
    public static string Response(string statement)
    {
        statement = statement.Trim();

        if (statement.IsAQuestion() && statement.IsShouting())
        {
            return "Calm down, I know what I'm doing!";
        }

        if (statement.IsAQuestion())
        {
            return "Sure.";
        }

        if (statement.IsShouting())
        {
            return "Whoa, chill out!";
        }

        if (statement.IsSilence())
        {
            return "Fine. Be that way!";
        }

        return "Whatever.";
    }

    private static bool IsAQuestion(this string statement) => statement.EndsWith('?');

    private static bool IsShouting(this string statement)
    {
        var statementLetters = Array.FindAll(statement.ToCharArray(), Char.IsLetter);

        return Array.TrueForAll(statementLetters, Char.IsUpper) && statementLetters.Length > 0;
    }

    private static bool IsSilence(this string statement) =>
        Array.TrueForAll(statement.ToCharArray(), Char.IsWhiteSpace);
}
